src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java

```diff
@@ -1,50 +1,46 @@
 package de.wejago.hichi2influx.repository;
 
 import com.influxdb.client.InfluxDBClient;
-import com.influxdb.client.WriteApiBlocking;
-import com.influxdb.client.domain.WritePrecision;
+import com.influxdb.client.WriteApi;
 import com.influxdb.client.write.Point;
 import de.wejago.hichi2influx.config.InfluxDBConfig;
-import de.wejago.hichi2influx.dto.SensorEntry;
-import java.time.Instant;
+import java.util.Queue;
+import java.util.concurrent.LinkedBlockingQueue;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Repository;
 
 @Repository
 @RequiredArgsConstructor
+@Slf4j
+@EnableScheduling
 public class InfluxDbRepository {
     private final InfluxDBConfig influxDBConfig;
-
     private InfluxDBClient influxDBClient;
+    private Queue<Point> measurementPoints = new LinkedBlockingQueue<>();
 
-    public Point writePoint(SensorEntry sensorEntry) {
-        updateConnectionIfNeeded();
-        WriteApiBlocking writeApi = influxDBClient.getWriteApiBlocking();
-
-        Point point = generateMeasurementPoint(sensorEntry);
-        writeApi.writePoint(point);
-
-        return point;
+    public void writePoint(Point point) {
+        measurementPoints.add(point);
     }
 
-    private void updateConnectionIfNeeded() {
-        //set the connection only if it is empty of not active
-        if(influxDBClient == null || !influxDBClient.ping()) {
-            influxDBClient = influxDBConfig.dbConnection();
+    @Scheduled(fixedRate = 30000)
+    public void persistPointsToInflux() {
+        log.info("persistPointsToInflux() - every 30 secs");
```

Comment:
Check if the queue is not empty, otherwise return. That saves time checking/creating DB connection.